### PR TITLE
fixed no schema matches error

### LINKED ISSUE #892: `No schema matches` error raised when key or value in a `json` record is a JSON array
When using the `json` embedded format and sending a message like the following ...

```json
{
  "records": [
    {
      "value": "my-value",
      "key": ["element1",{"field": "element2"}]
    }
  ]
}
```

The bridge returns ...

```json
{
    "error_code": 400,
    "message": "Validation error on: /records/0/key - No schema matches"
}
```

This error is raised by the internal Vert.x JSON schema component which is doing the validation on the `key` field based on the definition we have in the OpenAPI ...

```json
"key": {
    "oneOf": [
        {
            "type": "object"
        },
        {
            "type": "string"
        }
    ]
}
```

It should work because it's defined as a JSON object, and the array in the example is a valid JSON object.
So it seems that the array is making troubles to the validator.
The same happens if using the same to the `value` field of course.

### PR DESCRIPTION
**Pull request by antonio-pedro99**
This should fix #892.

I have added tests for both cases when the value or key is an array.

### CODE REVIEW

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java [hunk 1]
```diff
@@ -201,6 +201,109 @@ void sendSimpleMessageWithKey(VertxTestContext context) throws InterruptedExcept
...
+        consumer.handler(record -> {
+            context.verify(() -> {
+                assertThat(record.value(), is(value));
+                assertThat(record.topic(), is(topic));
+                assertThat(record.partition(), notNullValue());
+                assertThat(record.offset(), is(0L));
+                record.headers().forEach(header -> {
+                    assertThat(header.key(), is("key"));
+                    assertThat(header.value().toString(), is(array.encode()));
+                });
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:241]: I can't see any assert around the array you sent through the key field.

File: src/main/resources/openapi.json [hunk 2]
```diff
@@ -1897,6 +1909,9 @@
                     },
                     "key": {
                         "oneOf": [
+                            {
+                                "type": "array"
+                            },
```
Comment (ppatierno) [inline src/main/resources/openapi.json:1914]: All the above updates should be done in the openapiv2.json file as well.

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java [hunk 3]
```diff
@@ -201,6 +201,109 @@ void sendSimpleMessageWithKey(VertxTestContext context) throws InterruptedExcept
...
+        consumer.handler(record -> {
+            context.verify(() -> {
+                record.headers().forEach(header -> {
+                    assertThat(header.key(), is("value"));
+                    assertThat(header.value().toString(), is(value.encode()));
+                });
+                assertThat(record.topic(), is(topic));
+                assertThat(record.partition(), notNullValue());
+                assertThat(record.offset(), is(0L));
+                assertThat(record.key(), nullValue());
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:292]: as the previous one I don't see any check that value is the array you expected.

**Review by ppatierno — CHANGES_REQUESTED** [review]

**Comment by antonio-pedro99** [comment]
Hi @ppatierno, I will get back to you very soon with the updates

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java [hunk 4]
```diff
@@ -201,6 +201,103 @@ void sendSimpleMessageWithKey(VertxTestContext context) throws InterruptedExcept
...
+
+        KafkaConsumer<byte[], String> consumer = KafkaConsumer.create(vertx, consumerProperties,
+                new ByteArrayDeserializer(), new KafkaJsonDeserializer<>(String.class));
+        consumer.handler(record -> {
+            context.verify(() -> {
+                assertThat(record.value(), is(value));
+                assertThat(record.topic(), is(topic));
+                assertThat(record.partition(), notNullValue());
+                assertThat(record.offset(), is(0L));
+                assertThat(record.key(), is(key.toBuffer().getBytes()));
```
Comment (antonio-pedro99) [inline src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:238]: @ppatierno I am not sure if this is the best way to assert the key(which is in bytes). 
Any suggestions?

Comment (ppatierno) [reply src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:238]: The mistake here is totally different. Why did you delete the headers check? Kafka headers (with key-value) and key in the message are two totally different things. You should revert back the headers check part wherever you removed it.

**Review by ppatierno — CHANGES_REQUESTED** [review]

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java (hunk 4 above)
Comment (antonio-pedro99) [reply src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:238]: > The mistake here is totally different. Why did you delete the headers check? Kafka headers (with key-value) and key in the message are two totally different things. You should revert back the headers check part wherever you removed it.

Hmmm, I thought it would be better to follow other tests conventions. 
For example,  the `sendSimpleMessageWithKey(VertxTestContext context)` does not do any `header` check.

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,7 @@
 ## 0.29.0
 
 * Dependency updates (Vert.x 4.5.7, Netty 4.1.108.Final)
+* Fixed No schema matches error when using the `json` embedded format and the record key or value is an array.
```
Comment (ppatierno) [inline CHANGELOG.md:6]: I would reword and changing in terms of added support for key and value to be JSON array and not fixed bug imho.

File: src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java (hunk 4 above)
Comment (ppatierno) [reply src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java:238]: Yeah ok makes sense, even because we are not sending headers here as well.

**Review by ppatierno — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (antonio-pedro99) [reply CHANGELOG.md:6]: Maybe something like `Added support for key and value to be JSON array. Works with `json` embedded format.` would better?

Comment (ppatierno) [reply CHANGELOG.md:6]: Maybe a mix of that like "Added support for records key and value to be JSON array when using the `json` embedded format".

Comment (antonio-pedro99) [reply CHANGELOG.md:6]: Nice one!

Comment (antonio-pedro99) [inline CHANGELOG.md:6]:
```suggestion
* Added support for records key and value to be JSON array when using the json embedded format.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,7 @@
 ## 0.29.0
 
 * Dependency updates (Vert.x 4.5.7, Netty 4.1.108.Final)
+* Added support for records key and value to be JSON array when using the JSON embedded format.
```
Comment (ppatierno) [inline CHANGELOG.md:6]:
```suggestion
* Added support for records key and value to be JSON array when using the `json` embedded format.
```

**Review by ppatierno — CHANGES_REQUESTED** [review]

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]
